crates/sui-types/src/effects/object_change.rs

```diff
@@ -77,20 +85,30 @@ pub enum AccumulatorOperation {
 
 #[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
 pub enum AccumulatorValue {
-    // u64 should be sufficient for coin balance.
-    U64(u64),
+    Integer(u128),
```

Comment:
Why do we need u128 here?

Reply:
It's the only way to prevent overflow for Balance<T>. Unfortunately there is a loop hole that allows you to create Balance without a supply cap

Reply:
Oh, you were right about this btw. The u128 is needed at the storage layer, but not in effects. A transaction can only emit at most a u64.